x-pack/plugins/apm/common/agent_configuration/setting_definitions/index.ts

```diff
@@ -62,6 +63,19 @@ function getSettingDefaults(setting: RawSettingDefinition): SettingDefinition {
       };
     }
 
+    case 'storageSize': {
```

Comment:
How is `storageSize` different from `bytes`?

Reply:
`bytes` is intended for smaller sizes (e.g. header size limits), while `storageSize` is for disk space. `storageSize` type uses decimal units (3GB == 3 * 1000**3) while bytes uses the binary units ( 3mb == 3 * 1024**2). Also `storageSize` supports the larger units, while `bytes` only accepts up to the megabytes unit (`mb`). I started going down the path of allowing `bytes` to be more flexible to accept these differences by passing in options, but ended up creating it's own type due to the number of differences.

Reply:
It sounds like we should have `binaryByte` and `decimalByte`. That's the primary difference because the units are customisable.